AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -31,6 +31,16 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
 
     private let serialQueueSyncDeletions: DispatchQueue
 
+    private var modelName: ModelName
+
+    private let systemModelNames: Set = [MutationEvent.modelName,
+                                         ModelSyncMetadata.modelName,
+                                         MutationSyncMetadata.modelName]
+    
+    private var shouldLogDeveloperDefinedModel: Bool {
```

Comment:
Do we need to evaluate this on the fly every time it's accessed? 

It looks like `systemModelNames.contains(modelName)` can be evaluated once on initialization or lazily because `systemModelNames` and `modelName` are both immutable.

Do we need to support respecting a dynamically changed `log.logLevel` after the `CascadeDeleteOperation` is initialized? The cost of doing this is a blocking `nsLock.execute { }` on each `logLevel` `get` ([source](https://github.com/aws-amplify/amplify-swift/blob/e753770efddacd63125375a84fc8dc4c56ebc0ca/Amplify/Categories/Logging/LoggingCategory.swift#L45-L49)) , which seems less than ideal. Can we also evaluate that once with a lazy var?

i.e. 
```swift
private lazy var shouldLogDeveloperDefinedModel: Bool = { 
  log.logLevel >= .debug && Set([
    MutationEvent.modelName,
    ModelSyncMetadata.modelName,
    MutationSyncMetadata.modelName
  ]).contains(modelName)
}()
```

Reply:
On second thought, we can drop the `log.logLevel >= .debug` check here all together, can't we?

We're calling the applicable log level methods (e.g. `log.debug(...)`), so the logger internals will take care of that check.

Reply:
Assuming that's the case, we don't need `self` to do the evaluation anymore, so we can drop the `lazy` recommended above and define a local constant that's assigned in the `init`. Afaict, this also let's us drop the `modelName` property.

```swift
private let isDeveloperDefinedModel: Bool

init(...) {
  // ...

  self.isDeveloperDefinedModel = !Set([
    MutationEvent.modelName,
    ModelSyncMetadata.modelName,
    MutationSyncMetadata.modelName
  ]).contains(modelSchema.name)
}
```

Reply:
Updated to use jus the system models and a common log function